### PR TITLE
versions: upgrade flannel to 0.20.2 - main

### PR DESCRIPTION
**Pull request by ryansavino**
Upgrade flannel in versions.yaml

Fixes: https://github.com/kata-containers/tests/issues/5332

Signed-Off-By: Ryan Savino [ryan.savino@amd.com](mailto:ryan.savino@amd.com)

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @ryansavino!

**Review by wainersm — APPROVED** [review]
On CoCo Operator CI we are using a newer version of flannel (0.19). Anyway, I created an issue for us to remember to upgrade the version there too (https://github.com/confidential-containers/operator/issues/155).

Thanks @ryansavino !

**Comment by wainersm** [comment]
@ryansavino the jenkins-ci-ppc64le-ubuntu-20-04 job failed because the flannel container image was not pulled. It might be a intermittent problem on the node's network. Anyway, maybe we should run the CI jobs a couple of times to ensure this change doesn't not introduce a intermittent fail. 

```
INFO: Some expected Pods aren't running after 120 seconds.
NAMESPACE      NAME                            READY   STATUS                  RESTARTS   AGE
kube-flannel   kube-flannel-ds-dt8qx           0/1     Init:ImagePullBackOff   0          2m30s
```

File: integration/kubernetes/init.sh [hunk 1]
```diff
@@ -164,19 +164,19 @@ configure_network() {
 			info "Restart the CRI-O service due to $issue"
 			sudo systemctl restart crio
 		fi
-		local list_pods="kubectl get -n kube-system --selector app=flannel pods"
+		local list_pods="kubectl get -n kube-flannel --selector app=flannel pods"
```
Comment (cmaf) [inline integration/kubernetes/init.sh:167]: Hey @ryansavino, was testing this out to fix some of my own problems (https://github.com/kata-containers/tests/issues/5400) and I'm wondering if you also need to update kube-system on line 73?

**Comment by cmaf** [comment]
Should fix https://github.com/kata-containers/tests/issues/5400 as well

**Comment by wainersm** [comment]
Let me run the jobs again. If they all pass then we can merge it.

/test

**Review by cmaf — APPROVED** [review]

**Comment by wainersm** [comment]
The fail on ARM was cause by an unit test, probably a flaky unit test. Let me re-trigger it.

/test-arm-unit

**Comment by wainersm** [comment]
The failure of the s390x job is also an unrelated unit test.

/test-s390x-unit

**Comment by cmaf** [comment]
@wainersm could we merge or do you need those to pass?

**Comment by wainersm** [comment]
> @wainersm could we merge or do you need those to pass?

@cmaf the jenkins-ci-ppc64le-ubuntu-20-04 is still failing because it cannot pull the flannel image. I suppose it gets some pull limit to download from docker.io whereas flannel version 0.14 it was pulling from somewhere else. However I find no evidence of that on the logs...

Could you help here, @Amulyam24?  Mind if we merge this PR although it breaks ppc64le then we later try to fix it? Or do you prefer to have it fixed now? Asking because the problem we are trying to solve blocks other jobs.

**Comment by cmaf** [comment]
@ryansavino we actually have a bunch of ci jobs failing right now and the upgrade fixes them. hence trying to get it merged quickly 🙂

**Comment by Amulyam24** [comment]
> > @wainersm could we merge or do you need those to pass?
> 
> @cmaf the jenkins-ci-ppc64le-ubuntu-20-04 is still failing because it cannot pull the flannel image. I suppose it gets some pull limit to download from docker.io whereas flannel version 0.14 it was pulling from somewhere else. However I find no evidence of that on the logs...
> 
> Could you help here, @Amulyam24? Mind if we merge this PR although it breaks ppc64le then we later try to fix it? Or do you prefer to have it fixed now? Asking because the problem we are trying to solve blocks other jobs.

Hi @wainersm @cmaf. I'm okay with merging this. Please go ahead as this will help unblock the other jobs. I'll take a look and fix it on `ppc64le` later.

**Comment by Amulyam24** [comment]
> > > @wainersm could we merge or do you need those to pass?
> > 
> > 
> > @cmaf the jenkins-ci-ppc64le-ubuntu-20-04 is still failing because it cannot pull the flannel image. I suppose it gets some pull limit to download from docker.io whereas flannel version 0.14 it was pulling from somewhere else. However I find no evidence of that on the logs...
> > Could you help here, @Amulyam24? Mind if we merge this PR although it breaks ppc64le then we later try to fix it? Or do you prefer to have it fixed now? Asking because the problem we are trying to solve blocks other jobs.
> 
> Hi @wainersm @cmaf. I'm okay with merging this. Please go ahead as it is needed for the other jobs. I'll take a look and fix it on `ppc64le` later.

@wainersm  I looked into the issue and it is failing due to the image having no support for `ppc64le`
```
Warning  Failed     45s (x3 over 84s)  kubelet            Failed to pull image "docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0": rpc error: code = NotFound desc = failed to pull and unpack image "docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0": no match for platform in manifest: not found
```

There is an alternate images available but we might hit the docker rate limit with it
```
 - name: install-cni-plugin
       #image: flannelcni/flannel-cni-plugin:v1.1.0 for ppc64le and mips64le (dockerhub limitations may apply)
        image: docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0
```
Any idea or suggestions on how we might proceed with it?

**Comment by ryansavino** [comment]
> > > > @wainersm could we merge or do you need those to pass?
> > > 
> > > 
> > > @cmaf the jenkins-ci-ppc64le-ubuntu-20-04 is still failing because it cannot pull the flannel image. I suppose it gets some pull limit to download from docker.io whereas flannel version 0.14 it was pulling from somewhere else. However I find no evidence of that on the logs...
> > > Could you help here, @Amulyam24? Mind if we merge this PR although it breaks ppc64le then we later try to fix it? Or do you prefer to have it fixed now? Asking because the problem we are trying to solve blocks other jobs.
> > 
> > 
> > Hi @wainersm @cmaf. I'm okay with merging this. Please go ahead as it is needed for the other jobs. I'll take a look and fix it on `ppc64le` later.
> 
> @wainersm I looked into the issue and it is failing due to the image having no support for `ppc64le`
> 
> ```
> Warning  Failed     45s (x3 over 84s)  kubelet            Failed to pull image "docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0": rpc error: code = NotFound desc = failed to pull and unpack image "docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0": no match for platform in manifest: not found
> ```
> 
> There is an alternate images available but we might hit the docker rate limit with it
> 
> ```
>  - name: install-cni-plugin
>        #image: flannelcni/flannel-cni-plugin:v1.1.0 for ppc64le and mips64le (dockerhub limitations may apply)
>         image: docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.0
> ```
> 
> Any idea or suggestions on how we might proceed with it?

@Amulyam24 the blob from master (https://github.com/flannel-io/flannel/blob/3fa0ba795ff6f2b97a26c195850a12f40aa80c2b/Documentation/kube-flannel.yml#L131) uses:
image: `docker.io/flannel/flannel-cni-plugin:v1.1.2`

Can you check if that supports `ppc64le`? Maybe we just need to update again on the next flannel release?

**Pull request merged**